spec/clients/publishing_api/recommended_link_client_spec.rb

```diff
@@ -0,0 +1,91 @@
+RSpec.describe PublishingApi::RecommendedLinkClient, type: :client do
+  subject(:client) { described_class.new }
+
+  let(:recommended_link) { build(:recommended_link) }
+  let(:publishing_api_service) do
+    instance_double(GdsApi::PublishingApi, put_content: true, publish: true, unpublish: true)
+  end
+
+  before do
+    allow(GdsApi::PublishingApi).to receive(:new).with(
+      "https://publishing-api.test.gov.uk",
+      bearer_token: "test-token",
+    ).and_return(publishing_api_service)
+  end
+
+  describe "#create" do
+    it "puts and publishes the recommended link on Publishing API" do
+      expect(publishing_api_service).to receive(:put_content).with(
+        recommended_link.content_id,
+        recommended_link.to_publishing_api_content_item,
+      )
+
+      expect(publishing_api_service).to receive(:publish).with(recommended_link.content_id)
```

Comment:
gds-api-adapters also has test helper methods that could be used here: https://github.com/alphagov/gds-api-adapters/blob/main/lib/gds_api/test_helpers/publishing_api.rb#L251C11-L251C40